autofill/autofillScript.ts

```diff
@@ -1,18 +1,32 @@
+// Very basic function to find closest label to input field. Used for testing below
+function labelDist(labels: Element[], input: Element): Element {
+    let minDist = Number.MAX_VALUE;
+    let minLabel = labels[0];
+    for (let i = 0; i < labels.length; i++) {
+        let dist = Math.abs(labels[i].getBoundingClientRect().top - input.getBoundingClientRect().top);
+        if (dist < minDist) {
+            minDist = dist;
+            minLabel = labels[i];
+        }
+    }
+    return minLabel;
+}
+
+
 // This file will be run on the content page, and should include all the logic for inserting text into form items.
 const firstNameField = document.querySelector('input[name="First Name"]');
 const lastNameField = document.querySelector('input[name="Last Name"]');
 
 // Gets all input fields that contain keyword "email"
 const emailField = document.querySelectorAll('input[id*="email"]');
 
-//Figure out how to do a phone number field
 if (firstNameField) (firstNameField as HTMLInputElement).value = "yourFirstName";
 if (lastNameField) (lastNameField as HTMLInputElement).value = "yourLastName";
 
 // Sets all email fields to "yourEmail" as long as field ID contains "email"
-for (let i = 0; i < emailField.length; i++)
+/*for (let i = 0; i < emailField.length; i++)
     if (emailField[i]) (emailField[i] as HTMLInputElement).value = "yourEmail";
-//Figure out how to do a phone number field
+    */
 
 // Trigger agree to terms and conditions. Tested on "https://store.steampowered.com/join" but should work on any site with a checkbox with the word "agree" in the name. Proof of concept for checkboxes
 const agreeToTerms = document.querySelector('input[name*="agree"]');
```

Comment:
haha a robot agreeing to terms and conditions... we definitely might want to discuss whether we can really ethically do this. Maybe worth looking into a way to take the user to the fields that they must fill out manually for ethical reasons? Js has a really easy scrollTo method and we could highlight the field. Not necessary in this PR just something to think about

Reply:
you can also autocheck captchas too lol, but I thought that was a tad too far